### PR TITLE
Updateeth

### PR DESCRIPTION
**Pull request by jvlkl**
An Update for ethverein.

- Updated npm versions of multiple dependencies 

- Deleted unused dependencies with [depcheck](https://www.npmjs.com/package/depcheck) (mutiple dependencies got flagged as "unused", tough after the manual check most of them turned out to still be used, so be careful with this tool) 

- Updated Solidity and Truffle Code

- Added the possibility to launch ethverein with docker-compose

- Added hot module reload for the docker environment (Instant change after code edition, useful for development)

### CODE REVIEW

File: src/app/about/about.component.html [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+<p>&nbsp;</p>
+
+<h4>GitHub</h4>
+
+<a href="https://github.com/blocklab/ethverein">Find the Github Code for this Aplication here</a>
+
+<p>&nbsp;</p>
+
+<h4>Etherscan</h4>
+<a href="https://ropsten.etherscan.io/address/0x22ad32b66fACCB0b508d26B1FA57DA42a1DA9Db5#code">Find the Ethereum
```
Comment (jpbu) [inline src/app/about/about.component.html]: Shouldn't be hard-coded, we already have configuration properties `memberContractAddress` and `blockExplorerBaseUrl` for that.

File: src/app/about/about.component.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  selector: 'app-about',
+  templateUrl: './about.component.html',
+  styleUrls: ['./about.component.css']
+})
+
+export class AboutComponent implements OnInit {
+
+ 
+
+  displayedColumns = ['alias', 'status', 'block'];
```
Comment (jpbu) [inline src/app/about/about.component.ts]: Ist this stuff needed? Looks like a leftover from another component, please remove unnecessary properties or functions.

File: src/app/services/member-contract.service.ts [hunk 3]
```diff
@@ -35,161 +35,130 @@ export class MemberContractService {
...
+    this.membersContract.methods.setVotingContractAddress(_votingContractAdrres).send({from: acc})
   }
 
   /* /Contract Functions */
 
   /* Contract Events */
   getMemberAppliedEvent() {
-    return this.membersContract.MemberApplied();
+    return this.membersContract.events.MemberApplied({
+      filter: { myIndexedParam: [20, 23], myOtherIndexedParam: '0x123456789...' },
```
Comment (jpbu) [inline src/app/services/member-contract.service.ts]: Is this filter correct? Looks like sample data

File: src/build/contracts/Members.json [hunk 4]
```diff
@@ -0,0 +1,19739 @@
+{
```
Comment (jpbu) [inline src/build/contracts/Members.json]: `src/build` should be in .gitignore

File: src/truffle.js [hunk 5]
```diff
@@ -3,6 +3,11 @@ require('babel-register')
 
 module.exports = {
   networks: {
+    ganache:{
```
Comment (jpbu) [inline src/truffle.js]: I think this file can be removed when there is a `truffle-config.js`

File: src/app/services/member-contract.service.ts [hunk 6]
```diff
@@ -35,161 +35,130 @@ export class MemberContractService {
...
+      fromBlock: 0
+    }, function (error, event) { console.log(event); })
```
Comment (jpbu) [inline src/app/services/member-contract.service.ts]: Looks like the event is only logged, I think with the new web3.js version the function should take a callback as a parameter that does something useful with the event

**Review by jpbu — CHANGES_REQUESTED** [review]
See code comments, besides that it worked on my local machine including hot reload in the browser, nice work. I also pushed a commit with some slight adjustments to Dockerfile and docker-compose.

File: src/build/contracts/Members.json (hunk 4 above)
Comment (jvlkl) [reply src/build/contracts/Members.json]: should be or should not be? On my current version there is a src/build in the .gitignore file 

Screenshot from 2019-10-05 17-23-37

File: src/truffle.js (hunk 5 above)
Comment (jvlkl) [reply src/truffle.js]: Agreed :+1:

File: src/app/services/member-contract.service.ts (hunk 3 above)
Comment (jvlkl) [reply src/app/services/member-contract.service.ts]: fixed

File: src/app/about/about.component.ts (hunk 2 above)
Comment (jvlkl) [reply src/app/about/about.component.ts]: removed

File: src/build/contracts/Members.json (hunk 4 above)
Comment (jpbu) [reply src/build/contracts/Members.json]: The files in `src/build` should be ignored by git, however they are contained in the pull request. Looks like the gitignore is not working properly in this case.

Comment (jvlkl) [reply src/build/contracts/Members.json]: Seems I acidentally added those files in a commit. had to git rm the src/build folder.

.gitignore was/is working properly

File: src/app/services/member-contract.service.ts (hunk 6 above)
Comment (jvlkl) [reply src/app/services/member-contract.service.ts]: >  something useful with the event

Are you referring to something like a success message for the user?
e. g. "You(Member) successfully applied"

Comment (jpbu) [reply src/app/services/member-contract.service.ts]: There is already some frontend logic in place that refreshes certain content based on the events (e.g. update the member list of a member status changed). Best check what the previous version of the code did with the result of the events, and try to apply the same with the new/changed web3j library.

Comment (jvlkl) [reply src/app/services/member-contract.service.ts]: Resolved with: commit 82b09d21fdba4ef6683d8f5dd4b7af0f0f77bed7

**Pull request merged**